polkadot/runtime/rococo/src/lib.rs

```diff
@@ -2460,12 +2460,26 @@ sp_api::impl_runtime_apis! {
 	}
 
 	impl sp_genesis_builder::GenesisBuilder<Block> for Runtime {
-		fn create_default_config() -> Vec<u8> {
-			create_default_config::<RuntimeGenesisConfig>()
+		fn build_state(config: Vec<u8>) -> sp_genesis_builder::Result {
+			build_state::<RuntimeGenesisConfig>(config)
+		}
+
+		fn get_preset(id: &Option<PresetId>) -> Option<Vec<u8>> {
+			if let Some(id) = id {
+				genesis_config_presets::get_preset(id)
+			} else {
+				get_preset::<RuntimeGenesisConfig>(&None)
+			}
 		}
 
-		fn build_config(config: Vec<u8>) -> sp_genesis_builder::Result {
-			build_config::<RuntimeGenesisConfig>(config)
+		fn preset_names() -> Vec<PresetId> {
```

Comment:
minor annoyance that the list has to be duplicated between here and `genesis_config_preset.rs`, but I don't have a better solution off the top of my head either 🤷

Reply:
We could have a `Map` defined somewhere once. The preset_names would be `map.keys()` maybe.

Reply:
Yeah, it has to be a mutable static variable AFAIK.

Reply:
Maybe we could use [`async::Lazy`](https://docs.rs/once_cell/latest/once_cell/unsync/struct.Lazy.html) for this?

If that is fine for you, this could be taken to the follow-up PR. We either-way need to re-work the rest of runtimes so we can figure out something and fix rococo/substrate-test-runtime.